File: src/org/torproject/android/OnBootReceiver.java
package org.torproject.android;

import org.torproject.android.service.TorService;
import org.torproject.android.service.TorServiceUtils;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

public class OnBootReceiver extends BroadcastReceiver {
	
	@Override
	public void onReceive(Context context, Intent intent) {
		
		SharedPreferences prefs = TorServiceUtils.getSharedPrefs(context.getApplicationContext());
		
		boolean startOnBoot = prefs.getBoolean("pref_start_boot",true);
		
		if (startOnBoot)
		{
			startService("init",context);
			startService("start",context);			
		}

	}
	

	private void startService (String action, Context context)
	{
		
		Intent torService = new Intent(context, TorService.class);    
		torService.setAction(action);
		context.startService(torService);
		
        
	}
	
	
}

File: appcore/src/main/java/org/torproject/android/core/OnBootReceiver.kt
package org.torproject.android.core

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build

import org.torproject.android.service.OrbotConstants
import org.torproject.android.service.OrbotService
import org.torproject.android.service.util.Prefs

class OnBootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        try {
            if (Prefs.startOnBoot() && !sReceivedBoot) {
                //   if (isNetworkAvailable(context)) {
                startService(OrbotConstants.ACTION_START, context)
                sReceivedBoot = true
                // }
            }
        }
        catch (re: java.lang.RuntimeException) {
            //catch this to avoid malicious launches as document Cure53 Audit: ORB-01-009 WP1/2: Orbot DoS via exported activity (High)
        }
    }

    private fun startService(action: String, context: Context) {
        try {
            val intent = Intent(context, OrbotService::class.java).apply {
                this.action = action
            }.putNotSystem()
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
                context.startForegroundService(intent)
            else {
                context.startService(intent)
            }
        }
        catch (re: java.lang.RuntimeException) {
            //catch this to avoid malicious launches as document Cure53 Audit: ORB-01-009 WP1/2: Orbot DoS via exported activity (High)
        }
    }

    companion object {
        private var sReceivedBoot = false
    }
}